aop/src/components/file-uploader/FileUploader.vue

```diff
@@ -133,12 +135,26 @@ export default {
               return;
             }
             for (let pageNumber = 1; pageNumber <= pdfDoc.numPages; pageNumber++) {
+              const errorMessage = `Error reading page ${pageNumber} of the PDF.`;
               try {
-                const imageData = await this.getPage(pdfDoc, pageNumber);
-                const scaledImage = await this.scaleImage(imageData);
-                images.push(scaledImage);
-              } catch {
-                reject(`Error reading page ${pageNumber} of the PDF.`);
+                const imageSource = await this.getPage(pdfDoc, pageNumber);
+                const imageData = await this.getImageData(imageSource);
+                if (imageData.size > MAX_IMAGE_SIZE_BYTES) {
+                  imageData = await this.scaleImage(imageData);
```

Comment:
I think `imageData` should be defined as a `let` since there's another assignment occurring. In theory an error should occur right now for this case.

Reply:
Done, re-requested your review